### PR TITLE
feat: tooltip to show rejection reason for referrals 👤❌💬

### LINKED ISSUE #532: Rejection Reason Tooltip ❌
**Issue Reserved for Bloomberg Mentorship Program**

### Description

On the page ‘/profile/referrals’, members can provide ColorStack referrals. In the case that the referred individual is rejected, we want to provide the rejection reason. Members will be able to hover over the word ‘Rejected’, and then see a tooltip that describes the reason for rejection. 

### Acceptance Criteria

- Retrieve `rejection_reason` from the `applications` table
- Have this `rejection_reason` displayed as a tooltip on ‘/profile/referrals’ page in cases where referred person was rejected

### Additional Context

Take a look at where we implement this part of the UI on ‘/profile/referrals’ [here](https://github.com/colorstackorg/oyster/blob/4de42713dfee6bbfc2e07a19959d32b043e6734c/apps/member-profile/app/routes/_profile.profile.referrals.tsx#L81-L94). 

Note that to get the rejection reason into this file, we’ll have to work with the functions in [this](https://github.com/colorstackorg/oyster/blob/main/packages/core/src/modules/referral/referral.core.ts) file. Specifically, we want to be able to get the rejection reason for an application that has a specific `referral_id`. 

See an example of a tooltip implementation [here](https://github.com/colorstackorg/oyster/blob/main/apps/member-profile/app/routes/_profile.companies.tsx#L259).

### PR DESCRIPTION
**Pull request by wflore19**
## Description ✏️

Closes #532

- Allows users to view the specific reasons why certain referrals were rejected

rejection_reason_feature

## Type of Change 🐞

- [x] Feature - A non-breaking change which adds functionality.
- [ ] Fix - A non-breaking change which fixes an issue.
- [ ] Refactor - A change that neither fixes a bug nor adds a feature.
- [ ] Documentation - A change only to in-code or markdown documentation.
- [ ] Tests - A change that adds missing unit/integration tests.
- [ ] Chore - A change that is likely none of the above.

## Checklist ✅

- [x] I have done a self-review of my code.
- [x] I have manually tested my code (if applicable).
- [ ] I have added/updated any relevant documentation (if applicable).

### CODE REVIEW

**Comment by wflore19** [comment]
I'm going to double check and work on this with my Bloomberg mentor 📝

File: apps/member-profile/app/shared/utils/format.utils.ts [hunk 1]
```diff
@@ -25,3 +27,33 @@ export function formatName({
...
+ * @param {string} reason - The raw rejection reason.
+ * @returns {string} A formatted, user-friendly explanation of the rejection reason.
+ *
+ * @example
+ * // Returns "This referral was rejected because an ineligible major."
+ * formatRejectionReason('ineligible_major');
+ *
+ **/
+export function formatRejectionReason(reason: string) {
+  if (!reason) return 'Something happened.';
```
Comment (wflore19) [inline apps/member-profile/app/shared/utils/format.utils.ts:43]: Change to a more descriptive error.

File: apps/member-profile/app/shared/utils/format.utils.ts [hunk 2]
```diff
@@ -25,3 +27,33 @@ export function formatName({
...
+
+  switch (reason) {
+    case ApplicationRejectionReason.BAD_LINKEDIN:
+      return 'Incorrect or suspicious LinkedIn';
```
Comment (tomas-salgado) [inline apps/member-profile/app/shared/utils/format.utils.ts:47]: We might want to change the wording to just 'Incorrect LinkedIn' or maybe 'Incomplete LinkedIn'-using the word 'suspicious' could be problematic lol

Comment (tomas-salgado) [inline apps/member-profile/app/shared/utils/format.utils.ts:47]: We might want to change the wording to just 'Incorrect LinkedIn' or maybe 'Incomplete LinkedIn'-using the word 'suspicious' could be problematic lol

**Review by tomas-salgado — APPROVED** [review]
Great work @wflore19! I also tested this locally and it's working perfectly. Left one minor comment regarding wording but otherwise this LGTM!

**Comment by wflore19** [comment]
> Great work @wflore19! I also tested this locally and it's working perfectly. Left one minor comment regarding wording but otherwise this LGTM!

DancingHypeGIF

**Review by ramiAbdou — APPROVED** [review]
Great work, @wflore19! I updated the styling a bit (icon inside of the pill now), updated some copy and moved the formatting to the server.

Should hopefully get a lot less questions in the Slack w/ this!

Screenshot 2024-10-28 at 5 18 38 PM

**Comment by wflore19** [comment]
@ramiAbdou it looks better now, ty!

**Pull request merged**